hove-io/navitia
# Kraken: Fix Chaos after Kirin

:mag: Please review by commit and read message (the first 3 are "minor")

Realtime (Kirin) has priority over adapted (Chaos) when both are present.

Test:
1. Send Kirin then Chaos (rail-section), check realtime is Kirin's one
2. Send only Chaos (rail-section), check that realtime is Chaos' one

Fix (only 1. was failing):
When creating an adapted VJ, use the intersection of VP from disruption and the ValidityPattern (VP) of the original VJ: this is the "canceled_vp".
This canceled_vp is also used to deactivate other VJs of the meta-VJ.

This was OK for adapted VP, but the intersection from adapted was also used for realtime VP: this is wrong, as original VJ may already be deactivated for realtime (from previous Kirin's disruption).
This lead to wrongly deactivating Kirin's VJ.

Solution: process specific VP intersections for each RT-Level.
(doing the same for Kirin's case for code consistency).

JIRA: https://navitia.atlassian.net/browse/NAV-1925

Note that this bug was subject to order of disruptions' apply, and as applying a disruption triggers some other disruption's reapply, it lead to a bit random bug appearance.
